blocks/api/simple-dom.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * External dependencies
+ */
+import { forEach } from 'lodash';
+
+export function createSimpleNode( node, filter = Array ) {
+	if ( ! node ) {
+		return null;
+	}
+
+	if ( node.nodeType === 3 ) {
+		return node.nodeValue;
+	}
+
+	if ( node.nodeType !== 1 ) {
+		return null;
+	}
+
+	return filter(
+		node.nodeName.toLowerCase(),
+		Array.from( node.attributes || [] ).reduce( ( acc, { name, value } ) => ( {
+			...acc,
+			[ name ]: value,
+		} ), {} ),
+		...createSimpleNodeList( node.childNodes || [], filter )
+	);
+}
+
+export function createSimpleNodeList( nodeList, filter ) {
+	return Array.from( nodeList ).reduce( ( acc, node ) => {
+		const child = createSimpleNode( node, filter );
+
+		if ( Array.isArray( child ) && typeof child[ 0 ] !== 'string' ) {
+			acc.push( ...child );
+		} else if ( child ) {
+			acc.push( child );
+		}
+
+		return acc;
+	}, [] );
+}
+
+// Smarter application in the future?
+export function applySimpleNodeList( tree, node ) {
```

Comment:
I was initially confused on the naming here, though I think maybe this function could be more useful / understandable if it were to return a document fragment that could be used in Editable via `removeChild` + `appendChild`.

Reply:
Yeah, wasn't sure what the best way is, but went for appending right away. I thought we night eventually decide to create HTML from the tree and then set that instead. The nice thing about this is that it's fairly easy to create, and in the future we could do some DOM riffing to avoid updates where we can. I actually wonder if we there's nothing from React we can use there.